pywren_ibm_cloud/future.py

```diff
@@ -60,6 +60,7 @@ def __init__(self, call_id, callgroup_id, executor_id, activation_id, invoke_met
         self._new_futures = None
         self._traceback = None
         self._call_invoker_result = None
+        self._is_map_before_reduce = False
```

Comment:
@omerb01 With this, you can prevent get_results() method to download results by doing something like:
```
if self._is_map_before_reduce:
    return
```
in line 228